### PR TITLE
[1869] Document how to configure federation behind a proxy

### LINKED ISSUE #1869: Document Federation behind a proxy
## What Needs to be Documented?

With Server 10.3 two new config.php options have been added to enable federation when an instance needs to go through a proxy to reach a federated instance.

```
'proxy' => 'PROXY_HOSTNAME',
'proxyuserpwd' => 'USERNAME:PASSWORD',
```

- https://github.com/owncloud/core/pull/34101
- https://github.com/owncloud/enterprise/issues/3078

## Where Does This Need To Be Documented?

- Admin federation docs

## Why Should This Change Be Made? (Optional)

New feature

## What Type Of Content Change Is This? (Optional)
- [x] New Content Addition
- [ ] Old Content Deprecation
- [ ] Existing Content Simplification
- [ ] Bug Fix to Existing Content

## Which Manual Does This Relate To? (Optional)
- [x] Admin Manual
- [ ] Developer Manual
- [ ] User Manual
- [ ] Android
- [ ] iOS
- [ ] Branded Clients
- [ ] Desktop Client
- [ ] Other

### PR DESCRIPTION
**Pull request by settermjd**
This PR documents how to configure a federated instance to work through a proxy. It fixes #1869. For review, the commit to review is afc4a2e. 769ded8 is a clean-up commit and contains no changes directly pertinent to #1869.

:tipping_hand_man: **only backport to 10.3**, when approved.

### CODE REVIEW

**Comment by pmaier1** [comment]
THX! I'd like to ask @micbar for review if this is sufficient.

File: modules/admin_manual/pages/configuration/files/federated_cloud_sharing_configuration.adoc [hunk 1]
```diff
@@ -51,14 +53,18 @@ occ config:app:set federation auto_accept_trusted --value '0'
 occ config:app:set federatedfilesharing auto_accept_trusted --value 'yes'
 ----
 
+== Working With Proxies
+
+If the instance needs to go through a proxy, in `config/config.php`, set the `proxy` and `proxyuserpwd` configuration variables.
```
Comment (micbar) [inline modules/admin_manual/pages/configuration/files/federated_cloud_sharing_configuration.adoc]: I would like to have it better explained. There are oc instances that are not connected to the internet. They have no possibility to reach the public network. Therefore Federation will not work without a proxy. Some proxies require authentication.

Comment (settermjd) [reply modules/admin_manual/pages/configuration/files/federated_cloud_sharing_configuration.adoc]: Thanks for the feedback, @micbar. Did you accidentally approve the PR?

Comment (settermjd) [reply modules/admin_manual/pages/configuration/files/federated_cloud_sharing_configuration.adoc]: Can we talk about this on Friday?

Comment (micbar) [reply modules/admin_manual/pages/configuration/files/federated_cloud_sharing_configuration.adoc]: ok

Comment (settermjd) [reply modules/admin_manual/pages/configuration/files/federated_cloud_sharing_configuration.adoc]: Actually, can you clarify what you mean, here?

Comment (settermjd) [reply modules/admin_manual/pages/configuration/files/federated_cloud_sharing_configuration.adoc]: > I would like to have it better explained. There are oc instances that are not connected to the internet. They have no possibility to reach the public network. Therefore Federation will not work without a proxy. Some proxies require authentication.

I'm just seeking to find out how you want the explanation improved.

Comment (settermjd) [reply modules/admin_manual/pages/configuration/files/federated_cloud_sharing_configuration.adoc]: @micbar, I took your first comment, reworked it a little, and integrated it into the required section.

File: modules/admin_manual/pages/configuration/files/federated_cloud_sharing_configuration.adoc [hunk 2]
```diff
@@ -51,14 +53,22 @@ occ config:app:set federation auto_accept_trusted --value '0'
 occ config:app:set federatedfilesharing auto_accept_trusted --value 'yes'
 ----
 
+== Working With Proxies
+
+There are ownCloud instances that are not connected to the internet. 
+They have no possibility to reach the public network. 
+Therefore Federation will not work without a proxy. 
+And then there are others that need to go through proxies which require authentication.
```
Comment (micbar) [inline modules/admin_manual/pages/configuration/files/federated_cloud_sharing_configuration.adoc]: ok, I think the first 3 statements will be sufficient.

Comment (settermjd) [reply modules/admin_manual/pages/configuration/files/federated_cloud_sharing_configuration.adoc]: Updated.

**Pull request merged**